forc/src/ops/forc_init.rs

```diff
@@ -85,22 +108,67 @@ pub fn init(command: InitCommand) -> Result<()> {
 
     match command.template {
         Some(template) => {
-            let template_url = match template.as_str() {
-                "counter" => {
-                    Url::parse("https://github.com/FuelLabs/sway/tree/master/examples/hello_world")?
-                }
-                _ => {
-                    return Err(anyhow!(
-                        "Unrecognized template: \n Example Templates:\n - counter"
-                    ));
+            let example_url =
+                format!("https://github.com/FuelLabs/sway/tree/{SWAY_GIT_TAG}/examples/{template}");
+
+            let template_url = Url::parse(&example_url)?;
+
+            // Latest update to be changed into the match below
+            // init_from_git_template(project_name, &template_url).map_err(|e| anyhow!("Failed to initialize project from a template with the given name \"{template}\": {e}.\n  Note: If you are attempting to initialize this project from a Sway example, please ensure the template name matches one of the available examples.\n"))?;
+            // Ok(())
+
+            // If the user queried an existing example then continue otherwise attempt to fetch the examples and append them
+            // to the end of the error message so that the user can see the existing examples to choose from
+            match init_from_git_template(project_name, &template_url) {
+                Ok(()) => Ok(()),
+                Err(error) => {
+                    // TODO: change this to use anyhow!(), possibly without a match?
+                    let mut error_message = format!("Failed to initialize project from a template with the given name \"{template}\": {error}.\n  Note: If you are attempting to initialize this project from a Sway example, please ensure the template name matches one of the available examples.\n");
+                    let examples = get_sway_examples()?;
+                    for example in examples {
+                        error_message.push_str(format!("\t- {}\n", example).as_str());
+                    }
+                    println!("{}", error_message);
+                    Ok(())
```

Comment:
Rather than

```rust
                    println!("{}", error_message);
                    Ok(())
```
I think this could be (untested):

```rust
                    bail!("{}", error_message)
```